infra/general/kString.class.php

```diff
@@ -440,4 +440,9 @@ public static function stripInvalidUrlChars($url)
 		return str_replace(array('?', '|', '*', '\\', '/' , '>' , '<', '&', '[', ']',' '), '_', $url);
 	}
 
+	public static function keepOnlyValidUrlChars($url)
+	{
+	    return preg_replace('/[^A-Za-z0-9\-\.\_\~\!\$\&\(\)\*\+\,\;\=\:\@]/', '_', $url);
```

Comment:
Please keep only the needed slash, before the dash -, all the rest are not needed.